pkg/sql/create_table.go

```diff
@@ -1537,6 +1538,13 @@ func NewTableDesc(
 			if d.IsComputed() {
 				d.Computed.Expr = schemaexpr.MaybeRewriteComputedColumn(d.Computed.Expr, evalCtx.SessionData)
 			}
+			if d.GeneratedIdentity.IsGeneratedAsIdentity {
+				if !evalCtx.Settings.Version.IsActive(ctx, clusterversion.ExpressionIndexes) {
```

Comment:
ah, well these aren't expression indexes, so instead we should add a new clusterversion in pkg/clusterversion/cockroach_versions.go. you can make a new key here https://github.com/cockroachdb/cockroach/blob/65457f9df618177bfd56096c9be0194db2ef5cf8/pkg/clusterversion/cockroach_versions.go#L298 named `GeneratedAsIdentity` and add the version number for it here https://github.com/cockroachdb/cockroach/blob/65457f9df618177bfd56096c9be0194db2ef5cf8/pkg/clusterversion/cockroach_versions.go#L498

the idea is that once a cluster reaches that version key, it can't be downgraded back to an older version